bioengine/engine.py

```diff
@@ -0,0 +1,128 @@
+"""The BioEngine class."""
+import inspect
+import sys
+import yaml
+import os
+import threading
+import uuid
+from types import ModuleType
+
+from bioengine.utils import dotdict
+from bioengine.exceptions import UnsupportedAPI
+
+
+class BioEngine:
+    """The BioEngine class."""
+
+    def __init__(self, vendor_api=None):
+        """Set up engine."""
+        self.services = []
+        self.packages = {}
+        self._service_lock = threading.Lock()
+        self.vendor_api = vendor_api
+
+    def generate_engine_api(self, package_info=None, make_api_module=True):
+        engine_api = dotdict(showMessage=print, register=self.register_service)
+        if self.vendor_api:
+            engine_api.update(self.vendor_api)
+        if package_info:
+            engine_api.getPackageInfo = lambda: package_info
```

Comment:
If we need the camelcase names for imjoy support, we can register parallel api methods in snakecase.

Reply:
@oeway do we need to keep the camelCase names?

Reply:
Perhaps keep the camelcase for now?

Reply:
Ok. Then I'll add the parallel methods in snakecase too.